lib/mockingjay/strategies/gemm.ex

```diff
@@ -54,73 +54,48 @@ defmodule Mockingjay.Strategies.GEMM do
     mat_C = generate_matrix_C(trees, max_decision_nodes, max_leaf_nodes)
     {mat_D, mat_E} = generate_matrices_DE(trees, max_leaf_nodes, n_weak_learner_classes)
 
-    post_transform_args =
-      if opts[:post_transform] do
-        [custom_post_transform: opts[:post_transform]]
-      else
-        [n_classes: n_classes]
-      end
-
-    aggregate_args =
-      if opts[:aggregate] do
-        [custom_aggregate: opts[:aggregate]]
-      else
-        [
-          n_classes: n_classes,
-          n_trees: n_trees
-        ]
-      end
-
-    forward_args =
-      if opts[:forward] do
-        [custom_forward: opts[:forward]]
-      else
-        [
-          mat_A: mat_A,
-          mat_B: mat_B,
-          mat_C: mat_C,
-          mat_D: mat_D,
-          mat_E: mat_E,
-          condition: Mockingjay.Strategy.cond_to_fun(condition),
-          n_trees: n_trees,
-          max_decision_nodes: max_decision_nodes,
-          max_leaf_nodes: max_leaf_nodes,
-          n_weak_learner_classes: n_weak_learner_classes
-        ]
-      end
-
-    {forward_args, aggregate_args, post_transform_args}
+    arg = %{
+      mat_A: mat_A,
+      mat_B: mat_B,
+      mat_C: mat_C,
+      mat_D: mat_D,
+      mat_E: mat_E
+    }
+
+    opts = [
+      condition: Mockingjay.Strategy.cond_to_fun(condition),
+      n_trees: n_trees,
+      max_decision_nodes: max_decision_nodes,
+      max_leaf_nodes: max_leaf_nodes,
+      n_weak_learner_classes: n_weak_learner_classes,
+      n_classes: n_classes
+    ]
+
+    {arg, opts}
```

Comment:
I didn't apply this change to the other strategies, but ideally we would keep all tensors as arg, everything else as opts, to simplify handling inside the `forward` function.

Reply:
What's the benefit of having forward be a transform here if it's just validating keywords and then passing to a defn? I'm guessing you're using a map for args rather than keyword list since it's an Nx Container? I like that solution.

Reply:
There isn't really one. The important is to separate `{arg, opts}` as distinct arguments when calling `defn`. It can be done in several ways. One is forcing `init` to return a tuple and then you unpack the tuple and call `forward/3` directly.

Reply:
Ok, thanks for the clarification.  I think the pattern used here makes a lot of sense and is much easier to understand when writing as well. Thanks for updating it